src/borg/cache.py

```diff
@@ -452,25 +455,34 @@ def chunk_decref(self, id, stats):
         else:
             stats.update(-size, -csize, False)
 
-    def file_known_and_unchanged(self, path_hash, st, ignore_inode=False):
+    def file_cache_key(self, hash, path, st, ignore_inode):
+        if ignore_inode or fstype(path) not in FS_WITH_STABLE_INODES:
+            # we don't use the path directly (but its hash) to safe memory
+            cache_key = hash(safe_encode(path))
```

Comment:
ad 1: that's the price. can be avoided by either killing the files cache or using the env var to decrease kept generations.

ad 2: hash is a param of this function, not the builtin.